lib/concurrent/mvar.rb

```diff
@@ -68,10 +68,14 @@ def take(timeout = nil)
 
         # If we timed out we'll still be empty
         if unlocked_full?
-          value = @value
-          @value = EMPTY
-          @empty_condition.signal
-          apply_deref_options(value)
+          if block_given?
+            yield @value
```

Comment:
My objective is to be able to write code like this:
```ruby
f = MVar.new(IO.open('file', 'w'))
new_lines = f.take { |s| s.read }
new_lines = f.take { |s| s.read }
```
When using the block syntax, the user should not have to worry about putting back the value in the mvar.